src/dds/rsdds-serializable.cpp

```diff
@@ -0,0 +1,171 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2024 Intel Corporation. All Rights Reserved.
+
+#include "rsdds-serializable.h"
+
+#include <src/serialized-utilities.h>
+#include <src/core/sensor-interface.h>
+#include <src/core/device-interface.h>
+
+#include <rsutils/json.h>
+#include <set>
+
+using json = nlohmann::json;
+
+
+namespace librealsense {
+
+
+dds_serializable::dds_serializable()
+{
+}
+
+
+std::vector< uint8_t > dds_serializable::serialize_json() const
+{
+    serialized_utilities::json_preset_writer writer;
+    writer.set_device_info( get_serializable_device() );
+
+    // Set of options that should not be written
+    std::set< rs2_option > const options_to_ignore{
+        RS2_OPTION_FRAMES_QUEUE_SIZE,  // Internally added and is not an option we need to record/load
+        RS2_OPTION_REGION_OF_INTEREST  // The RoI is temporary, and we don't load it
+    };
+
+    for( auto p_sensor : get_serializable_sensors() )
+    {
+        std::string const sensor_name = p_sensor->get_info( RS2_CAMERA_INFO_NAME );
+        for( auto o : p_sensor->get_supported_options() )
+        {
+            auto & opt = p_sensor->get_option( o );
+            if( opt.is_read_only() )
```

Comment:
If option is read only we don't want to output it?

Reply:
We don't want to write it.
Imagine writing the temperatures -- makes no sense.